### ISSUE TITLE
invalid exported symbol collision for _start symbol

### ISSUE THREAD

**Issue by emekoi** [ziglang/zig#1691]
allow the user to define a custom `bootstrap.zig` file or access the root file through `@root` like in `bootstrap.zig`.

**Comment by andrewrk** [comment]
Can you expand on your use case a little bit?

I think this already works. If zig detects that you provided the entry point symbol for a given target, it omits bootstrap.zig.

**Comment by andrewrk** [comment]
OK, this is a bug:

```
/home/andy/downloads/zig/build/lib/zig/std/special/bootstrap.zig:17:9: error: exported symbol collision: '_start'
        @export("_start", _start, strong_linkage);
        ^
```

Zig is supposed to detect that this is an entry symbol and  omit bootstrap.zig. Here's the workaround:

```
zig build-obj foo.zig
zig build-exe --name foo --object foo.o
```

**Comment by Tetralux** [comment]
Would it make sense to be able to mark an arbitrary function as the entry point, have a cmdline flag, or `build.zig` option for it? Provided it has the right signature, of course.

**Comment by andrewrk** [comment]
This is now solved (by #2966) with a userland implementation. I think the limitation that the `_start` function has to be in the root source file is reasonable.
